common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -659,7 +697,17 @@ function (Sjson, AsyncProcess) {
                 }
 
                 time = Math.round(time * 1000 + 100);
-                newIndex = this.sjson.search(time);
+                var startTimeInSec = state.config.startTime,
+                    endTimeInSec = state.config.endTime;
+                // if start and end times are defined, limit search.
+                // else, use the entire list of video captions
+                if (endTimeInSec !== null) {
```

Comment:
you seem to have lots of code repetition in this file, any way to make a helper function and just get that to repeat code for you?